modules/openy_gc_personal_training/openy_gc_personal_training.install

```diff
@@ -122,3 +122,59 @@ function openy_gc_personal_training_update_8004(&$sandbox) {
     }
   }
 }
+
+/**
+ * Add "State" field to "Personal Training" entity.
+ */
+function openy_gc_personal_training_update_8005($sandbox) {
+  $state_field_storage_definition = BaseFieldDefinition::create('state')
+    ->setLabel(t('State'))
+    ->setDescription(t('The personal training state.'))
+    ->setRequired(TRUE)
+    ->setSetting('max_length', 255)
+    ->setSetting('workflow', 'personal_training_default')
+    ->setDefaultValue('planned')
+    ->setDisplayOptions('form', [
+      'type' => 'options_select',
+    ])
+    ->setDisplayConfigurable('form', TRUE)
+    ->setDisplayConfigurable('view', TRUE);
+
+  \Drupal::entityDefinitionUpdateManager()
+    ->installFieldStorageDefinition(
+      'state',
+      'personal_training',
+      'personal_training',
+      $state_field_storage_definition
+    );
+}
+
+/**
+ * Revoke permission "delete personal training entities" from virtual trainer.
+ */
+function openy_gc_personal_training_update_8006(&$sandbox) {
+  $configuration = \Drupal::configFactory()->getEditable('user.role.virtual_trainer');
+  $permissions = $configuration->get('permissions');
+
+  $index = array_search('delete personal training entities', $permissions, TRUE);
+  if ($index !== FALSE) {
+    unset($permissions[$index]);
+    $configuration->set('permissions', $permissions)->save();
+  }
+}
+
+/**
+ * Move meeting_delete_* params to meeting_cancel_*.
+ */
+function openy_gc_personal_training_update_N(&$sandbox) {
```

Comment:
Why do we have _N at the end. It seems to me that it will never be executed.
What is the purpose of the hook_update?

Reply:
Sorry, i didn't replace `N` with `8007`. The purpose of the hook is to move values within `openy_gc_personal_training.settings` config to the different keys.